client/core/core.go

```diff
@@ -943,6 +943,44 @@ func (c *Core) Login(pw []byte) ([]*db.Notification, error) {
 	return notes, nil
 }
 
+// Logout logs the user out
+func (c *Core) Logout() error {
+	c.connMtx.RLock()
+	defer c.connMtx.RUnlock()
+
+	// Check active orders and lock wallets
+	for assetID := range c.user.Assets {
+		for _, dc := range c.conns {
+			if dc.hasOrders(assetID) {
+				return fmt.Errorf("cannot log out with active orders, wallet %s ", unbip(assetID))
+			}
+		}
+		wallet, found := c.wallet(assetID)
+		if found && wallet.connected() {
+			if err := wallet.Lock(); err != nil {
+				return err
+			}
+		}
+	}
+
+	// Lock dex accounts
+outer:
+	for _, dc := range c.conns {
+		if dc.acct.locked() {
+			continue outer
+		}
+		for assetID := range dc.assets {
+			if _, found := c.user.Assets[assetID]; found {
+				dc.acct.lock()
+				continue outer
+			}
+		}
+	}
```

Comment:
Why only lock the dexAccount if at least one of the client's supported assets is used by that dex?

It seems to me that this could just be:

```go
// With no open orders for any of the dex connections, and all wallets locked,
// lock each dex account.
for _, dc := range c.conns {
	dc.acct.lock()
}
```